### DISCUSSION TITLE
How do I change the size of this window to be fullscreen, or add the system title bar?

### DISCUSSION THREAD

**Discussion by GEOEGII555** [microsoft/wslg#1286] [Q&A]
I have this wayland compositor borderless window:
![Wayland compositor window](https://github.com/user-attachments/assets/7a37e397-3906-4d47-a6c1-1631d1e25808)
How do I make it fullscreen or add a title bar to it?

**Comment by crramirez** [comment]
Share the command you used to do this. Are you using Xwayland?

**Reply by GEOEGII555** [reply]
`startplasma-wayland`

**Reply by crramirez** [reply]
Try this: `Xwayland -terminate -decorate :1 & WAYLAND_DISPLAY= DISPLAY=:1 startplasma-x11` probably you'll need to install the x11 support for plasma. For more info: https://gist.github.com/tdcosta100/e28636c216515ca88d1f2e7a2e188912#gistcomment-5199541

**Reply by crramirez** [reply]
for full screen: `Xwayland -terminate -fullscreen :1 & WAYLAND_DISPLAY= DISPLAY=:1 startplasma-x11` to exit press Alt+F4

**Reply by GEOEGII555** [reply]
> `Xwayland -terminate -fullscreen :1 & WAYLAND_DISPLAY= DISPLAY=:1 startplasma-x11`

It tries to start, but I just see a black screen with the only sign of Plasma being the cursor.

EDIT: it started, but it's restarting in a loop: `Application::crashHandler() called with signal 11; recent crashes: 4`, there's no background and windows don't have borders

**Comment by beedell-roke** [comment]
> https://github.com/microsoft/wslg/discussions/1286#discussion-7203531

@GEOEGII555, it looks to me like the reason why you see no title bar is `plasmashell` (in `plasma-workspace-6.3.2-1.fc41.x86_64`) renders itself with, in `.kwinrule` syntax:

~~~desktop
noborder=true
noborderrule=2
~~~

This might be remediable by:

1. Visiting `kcm_kwinrules` (via `systemsettings` or `kcmshell6`);
2. Importing the undermentioned:

   ~~~desktop
   [Placeholder]
   Description=Placeholder
   noborderrule=2

   ~~~

Ultimately, the undermentioned should work with XOrg 11:

![Screenshot_20250309_182038](https://github.com/user-attachments/assets/38f4ccd4-e8fd-4a40-ad1a-93560cc35c09)

However, if it's too broad, utilise <kbd>Alt</kbd>+<kbd>F3</kbd> to refine it by looking at the window class and inputting it into the KCM's GUI:

![Screenshot_20250309_182236](https://github.com/user-attachments/assets/66250a89-13fe-473d-bf76-a5337c42aaad)

This probably shan't work under Wayland due to [`bugs.kde.org/show_bug.cgi?id=452240`](https://bugs.kde.org/show_bug.cgi?id=452240#c0:~:text=%22No%20titlebar%20and%20frame%22%20rule%20doesn't%20force%20SSDs%20on%20Wayland.).
